homeassistant/components/smtp/__init__.py

```diff
@@ -1,6 +1,100 @@
 """The smtp component."""
+from __future__ import annotations
 
-from homeassistant.const import Platform
+import logging
+import smtplib
+import socket
+from typing import Any
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_TIMEOUT,
+    CONF_USERNAME,
+    CONF_VERIFY_SSL,
+    Platform,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+from homeassistant.helpers import discovery
+from homeassistant.helpers.typing import ConfigType
+from homeassistant.util.ssl import client_context
+
+from .const import CONF_DEBUG, CONF_ENCRYPTION, CONF_SERVER, DATA_HASS_CONFIG, DOMAIN
 
-DOMAIN = "smtp"
 PLATFORMS = [Platform.NOTIFY]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up the smtp component."""
+
+    hass.data[DATA_HASS_CONFIG] = config
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up smtp from a config entry."""
+
+    try:
+        smtp_client = await hass.async_add_executor_job(
+            get_smtp_client, dict(entry.data)
+        )
+        smtp_client.quit()
+
+    except (socket.gaierror, ConnectionRefusedError) as err:
+        _LOGGER.error(
+            "SMTP server not found or refused connection (%s:%s). "
+            "Please check the IP address, hostname, and availability of your SMTP server",
+            entry.data[CONF_SERVER],
+            entry.data[CONF_PORT],
+        )
+        raise ConfigEntryNotReady from err
+
+    except smtplib.SMTPAuthenticationError as err:
+        raise ConfigEntryAuthFailed(
+            "Login rejected. Please check your credentials"
+        ) from err
+
+    hass.async_create_task(
+        discovery.async_load_platform(
+            hass,
+            Platform.NOTIFY,
+            DOMAIN,
+            dict(entry.data),
+            hass.data[DATA_HASS_CONFIG],
+        )
+    )
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    return await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
```

Comment:
I believe we need to pop the entry from hass.data like in the scaffold?
https://github.com/home-assistant/core/blob/f29cc33fa0652b3d1c1c71081344d7b5b80e174a/script/scaffold/templates/config_flow/integration/__init__.py#L25-L30

Reply:
We are not storing the entry in hass.data so there is nothing to pop.

Reply:
True.  Should be fine then.